File: 0128-longest-consecutive-sequence/0128-longest-consecutive-sequence.java
class Solution {
    public int longestConsecutive(int[] nums) {
        // construct min heap
        PriorityQueue<Integer> heap = new PriorityQueue<>();
        for (int num : nums)
            heap.add(num);
        
        int len, res = 0;
        while (!heap.isEmpty()) {
            int left = heap.poll();
            len = 1;
            while (!heap.isEmpty() && heap.peek() - left <= 1) {
                if (heap.peek() - left == 1)
                    len++;
                left = heap.poll();
            }
            res = Math.max(res, len);
        }
        
        return res;
    }
}

File: 0128-longest-consecutive-sequence/0128-longest-consecutive-sequence.py
class Solution:
    def longestConsecutive(self, nums: List[int]) -> int:
        res = 0
        hashset = set(nums)
        
        for n in hashset:
            if n - 1 not in hashset:
                curr = n
                count = 1
                while curr + 1 in hashset:
                    curr += 1
                    count += 1
                res = max(count, res)
        
        return res
            